### DISCUSSION TITLE
Using D1, returning a nested JSON object as an object instead of a string? Or fully expanding a deep JSON object

### DISCUSSION THREAD

**Discussion by exoup** [cloudflare/workers-sdk#5695] [Q&A] [answered]
Using a D1 database and given the following object as input:

```json
{
        "id": "2rQAni2KRIW_2L3OQDnrc",
        "images": {
            "full": "",
            "thumbnail": "",
            "foilMask": "",
            "local": "2rQAni2KRIW_2L3OQDnrc"
        },
        "flavorText": "“We know someone who can help you...for a price.”",
        "abilities": [
            "Rush (This character can challenge the turn they're played.)"
        ],
        "effects": [
            {
                "name": "DEXTEROUS LUNGE",
                "text": "Your characters named Jetsam gain Rush."
            }
        ],
        "subtypes": [
            "Storyborn",
            "Ally"
        ]
    }
```

I use `json_extract `to extract the json in my response, but only the top-level values are returned appropriately.
Instead, for instance with images, I receive a text response like this:
`'{"full":"","thumbnail":"","foilMask":"","local":"2rQAni2KRIW_2L3OQDnrc"}'`

Currently I'm using a SQL query like this: `SELECT json_extract(jsonText, '$') FROM Table ORDER BY random() LIMIT 1` just for testing.

What do I need to do in order to return the nested text as an object?

I see this same issue when creating generated columns like `images AS (json_extract(jsonText, '$.images')) STORED,`

**Comment by treeder** [comment, marked as answer]
```js
let object = await d1....
object.images = JSON.parse(object.images)
```
